### DISCUSSION TITLE
Combing ripgrep with git blame output? How to prefilter matches through git blame?

### DISCUSSION THREAD

**Discussion by gcflymoto** [BurntSushi/ripgrep#2262] [Q&A] [answered]
There is a way to do this with 

```
git grep -n content  | perl -F':' -anpe '$_=`git blame -L$F[1],+1 $F[0]`'

and

rg -n content | perl -F':' -anpe '$_=`git blame -L$F[1],+1 $F[0]`'
```
as documented here
https://stackoverflow.com/questions/28724057/git-blame-of-particular-lines-inside-all-files-filtered-with-grep-command 

but we lose all the benefits of the excellent output rg provides when it is not piping.

**Comment by BurntSushi** [comment]
If there was a way, it would probably be with some creative use of the `--pre` flag. But I'm not a Perl user and your command here looks pretty opaque to me. So this question isn't terribly accessible to me as written.

**Comment by gcflymoto** [comment, marked as answer]
To reduce the cost of running "git blame" on every file before matching which is costly, this double search method provides the desired output
```
rg --pre git-blame content `rg -l content`
```
@BurntSushi thanks for the advice

**Reply by BurntSushi** [reply]
Very cool! Nice and simple. :)

**Reply by BurntSushi** [reply]
Thank you for posting the answer that worked for you!
